lib/shelley/test/integration/Main.hs

```diff
@@ -321,3 +337,53 @@ minSeverityFromEnv def var = lookupEnv var >>= \case
     Nothing -> pure def
     Just "" -> pure def
     Just arg -> either die pure (parseLoggingSeverity arg)
+
+-- | Pre-register a staking key for the STAKE_POOLS_JOIN_05 test.
+preregisterStakingKeysForTests :: NetworkLayer IO t b -> IO ()
+preregisterStakingKeysForTests nl = do
```

Comment:
I am not fond of having this function here but I guess it'll do for now. Ideally, I'd have put that where the rest of the "setup & fixture" work is happening (i.e. in the "Launch" module).